influxvar.go

```diff
@@ -41,5 +53,39 @@ func NewStatistics(key, name string, tags map[string]string) *expvar.Map {
 	statMap.Init()
 	m.Set("values", statMap)
 
+	// Create a reference counter that we can use to release the map.
+	referencesVar := &expvar.Int{}
+	referencesVar.Set(2) // once for the site of usage, once for the monitor
+	m.Set("references", referencesVar)
 	return statMap
 }
+
+// Iterate over all the statistics maps.
+func DoStatistics(fn func(expvar.KeyValue)) {
+	root.Do(fn)
+}
+
+// Used to deregister a statistic when it is no longer needed.
+func DeleteStatistics(key string) {
```

Comment:
I agree this is an annoyance. 

My primary objective with this change was to solve the leak and I wanted to do so in a way that minimises the invasiveness of the change to the rest of the code base. 

I'd be happy to work on eliminating that annoyance in a subsequent PR which might need to drag in a wide range of interested parties. That said, if you think it needs to be addressed now, let me know.